lib/SIL/SILVerifier.cpp

```diff
@@ -751,7 +751,8 @@ class SILVerifier : public SILVerifierBase<SILVerifier> {
 
     // If there are substitutions, verify them and apply them to the callee.
     if (subs.empty()) {
-      require(!fnTy->isPolymorphic(),
+      require(!fnTy->isPolymorphic() ||
+                  fnTy->getGenericSignature()->areAllParamsConcrete(),
```

Comment:
The former. If all generic parameters were made concrete in a signature, just drop the signature when the SILFunctionType is constructed. We already do this for methods defined in concrete extensions (e.g. extension Array where Element == Int) so I'd greatly prefer to also do that for full specializations instead of adding a new possible state for SILFunctionType.